Add snap line to more angles (fix #1641)

Gasparoken opened the pull request:
This PR is a cherry-pick and updates the original PR [#4241](https://github.com/aseprite/aseprite/pull/4241) from [tsone](https://github.com/tsone)
fix #1641

tsone committed:
Add snap line to more angles (fix #1641)

martincapello reviewed (dismissed):
It works great! The only thing that I was just thinking about is that the last step is the only step that might not be the same size as the others...so I was wondering if we should force the last step size to be the same as the others, so the line would always have all their steps of the same size...just something to think about. Anyway I'm approving this!

martincapello reviewed (commented):
Gave a second look to the changes (because Gaspar made me notice about the static variable in algo.h). So I think we should remove the `fixed_step_tilt` static variable.

Gasparoken committed:
Add Fixed step tilt preference for editor behavior with the Line tool

Before this and the previous commit, the Line tool allowed drawing
lines with a regular stair-like pattern at 45° and 26.56°
(using only 2-pixel horizontal segments) via the SHIFT modifier.
Now, the option to generate these fixed lines up to 256 pixels
has been added to the preferences (2 as default).

dacap commented:
I have just rebased this to latest beta in https://github.com/aseprite/aseprite/tree/snap-to-more-angles

I've also added the possibility to just leave one angle (45 degrees). I didn't make any further review of the PR, you can reset your branch to snap-to-more-angles and I'll merge it and then see if the configuration UI can be improved a little (e.g. a slider instead of a text entry field).
